google-cloud-core/src/main/java/com/google/cloud/ServiceOptions.java

```diff
@@ -212,6 +216,10 @@ public B setCredentials(Credentials credentials) {
       if (this.projectId == null && credentials instanceof ServiceAccountCredentials) {
         this.projectId = ((ServiceAccountCredentials) credentials).getProjectId();
       }
+
```

Comment:
It seems surprising to fail if the credentials are already set; that is, not to allow resetting them.
If that's intended, it should be documented in the JavaDoc.

Reply:
I am not sure I follow this defaults to the quota project from GoogleCredentials(Application default) if the service option is not set. If the user manually provided a ServiceOption to set this, that option takes precedence.

Reply:
This can be done separately: https://github.com/googleapis/java-core/issues/96